.github/workflows/tests.yml

```diff
@@ -1,17 +1,74 @@
-name: tests
+name: "Build and Test"
 
 on: [push, pull_request]
 
 jobs:
-  test:
+  build-and-test:
     runs-on: ubuntu-latest
     steps:
+    - uses: actions/checkout@v2
+
     - name: Install Deno
       run: curl -fsSL https://deno.land/x/install/install.sh | sh
-    - uses: actions/checkout@v1
+
+    - name: Build generated files
+      id: build
+      run: |
+        export DENO_INSTALL="/home/runner/.deno"
+        export PATH="$DENO_INSTALL/bin:$PATH"
+        rm -f build/sqlite.{js,wasm} build/lib/sqlite3.{c,h}
+        sudo apt-get -y install --no-install-recommends tclsh gcc
+        cd build
+
+        make setup
+        make amalgamation
+        make release
+
+        git status
+        declare had_changes="$(
```

Comment:
I think it might be worth looking into https://github.com/stefanzweifel/git-auto-commit-action for this? This is a lot of bash script here, and I think using that action we can simplify this to something like what he suggests in his README:

```yaml
on:
  pull_request:
  push:
    branches:
      - "main"

jobs:
  build-and-test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
      with:
        ref: ${{ github.head_ref }}

    - name: Install Deno
      // etc...

    - uses: stefanzweifel/git-auto-commit-action@v4
      with:
        commit_message: (GitHub Action) Rebuild SQLite WASM
```

Reply:
(I believe (but we need to test this), that this should also work on PRs/ that there is some way to make it work on PRs)

Reply:
done